fri/server/main.py

```diff
@@ -114,33 +79,75 @@ def build(dir):
 
 @app.route('/debug/<dir>', methods=['POST'])
 def debug(dir):
-    cur_path = os.getcwd()
-    concore_path = os.path.abspath(os.path.join(cur_path, '../../'))
+    dir = secure_filename(dir)
     dir_path = os.path.abspath(os.path.join(concore_path, dir))
     proc = call(["./debug"], cwd=dir_path)
     if(proc == 0):
-        resp = jsonify({'message': 'Close the pop window after obtaing result'})
+        resp = jsonify({'message': 'Close the pop window after obtaining result'})
         resp.status_code = 201
         return resp
     else:
         resp = jsonify({'message': 'There is an Error'})
         resp.status_code = 500
         return resp  
 
-# to download /download/<dir>?fetch=<downloadfile>. For example, /download/test?fetch=example.py.out&apikey=xyz
+
+@app.route('/run/<dir>', methods=['POST'])
+def run(dir):
+    dir = secure_filename(dir)
+    dir_path = os.path.abspath(os.path.join(concore_path, dir))
+    proc = call(["./run"], cwd=dir_path)
+    if(proc == 0):
+        resp = jsonify({'message': 'result prepared'})
+        resp.status_code = 201
+        return resp
+    else:
+        resp = jsonify({'message': 'There is an Error'})
+        resp.status_code = 500
+        return resp
+
+@app.route('/stop/<dir>', methods=['POST'])
+def stop(dir):
+    dir = secure_filename(dir)
+    dir_path = os.path.abspath(os.path.join(concore_path, dir))
+    proc = call(["./stop"], cwd=dir_path)
+    if(proc == 0):
+        resp = jsonify({'message': 'resources cleaned'})
+        resp.status_code = 201
+        return resp
+    else:
+        resp = jsonify({'message': 'There is an Error'})
+        resp.status_code = 500
+        return resp                
+                   
+
+@app.route('/clear/<dir>', methods=['POST'])
+def clear(dir):
+    dir = secure_filename(dir)
+    dir_path = os.path.abspath(os.path.join(concore_path, dir))
+    proc = call(["./clear"], cwd=dir_path)
+    if(proc == 0):
+        resp = jsonify({'message': 'result deleted'})
+        resp.status_code = 201
+        return resp
+    else:
+        resp = jsonify({'message': 'There is an Error'})
+        resp.status_code = 500
+        return resp
+
+# to download /download/<dir>?fetch=<downloadfile>. For example, /download/test?fetchDir=xyz&fetch=u
 @app.route('/download/<dir>', methods=['POST', 'GET'])
 def download(dir):
     download_file = request.args.get('fetch')
-    apikey = request.args.get('apikey')

```

Comment:
Why are you removing apikey here?

Reply:
we are downloading file from sample1 folder. which is created when we call build and debug.